jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/Layers.java

```diff
@@ -0,0 +1,179 @@
+
+          // relativize all paths to the buildRoot location
+          Path rawSrc = copySpec.getSrc();
+          Path src = rawSrc.isAbsolute() ? rawSrc : buildRoot.resolve(rawSrc);
+          AbsoluteUnixPath dest = copySpec.getDest();
+
+          if (Files.isRegularFile(src)) { // regular file
+            if (copySpec.getExcludes().size() > 0 || copySpec.getIncludes().size() > 0) {
+              throw new UnsupportedOperationException(
+                  "Cannot apply includes/excludes on single file copy directives.");
+            }
+            AbsoluteUnixPath target = dest;
+            if (copySpec.isDestEndsWithSlash()) {
+              target = copySpec.getDest().resolve(src.getFileName());
+            }
+            layerBuiler.addEntry(
+                src,
+                target,
+                filePropertiesStack.getFilePermissions(),
+                filePropertiesStack.getModificationTime(),
+                filePropertiesStack.getOwnership());
+          } else if (Files.isDirectory(src)) { // directory
+            List<PathMatcher> excludes =
+                copySpec
+                    .getExcludes()
+                    .stream()
+                    .map(
+                        pattern ->
+                            pattern.endsWith("/") || pattern.endsWith("\\")
+                                ? pattern + "**"
+                                : pattern)
+                    .map(exclude -> FileSystems.getDefault().getPathMatcher("glob:" + exclude))
+                    .collect(Collectors.toList());
+            List<PathMatcher> includes =
+                copySpec
+                    .getIncludes()
+                    .stream()
+                    .map(
+                        pattern ->
+                            pattern.endsWith("/") || pattern.endsWith("\\")
+                                ? pattern + "**"
+                                : pattern)
+                    .map(include -> FileSystems.getDefault().getPathMatcher("glob:" + include))
+                    .collect(Collectors.toList());
+            try (Stream<Path> dirWalk = Files.walk(src)) {
+              dirWalk
+                  .filter(
+                      (Predicate<Path>)
+                          path -> {
+                            // filter out against excludes
+                            for (PathMatcher matcher : excludes) {
+                              if (matcher.matches(path)) {
+                                return false;
+                              }
+                            }
+                            // if there are no includes directives, include everything
+                            if (includes.isEmpty()) {
+                              return true;
+                            }
+                            // TODO: for directories that fail to match the "include" directive on
+                            // TODO: files, populate the directories somehow or just never apply
```

Comment:
could you elaborate more on this or provide an example?

Reply:
hrmm.. yeah lemme go update this comment